prime-router/docs/design/validation/design.md

```diff
@@ -0,0 +1,286 @@
+This step will not be implemented for HL7
+
+#### Post sender transform
+Will occur during the convert function, after receiving and debatch, and before the convert step. We again need to 
+ensure that all necessary data exists and is sane. This will happen for both HL7 and FHIR. One final validation will be 
+needed (per receiver) after applying receiver transforms and enrichment.
+
+#### Post receiver enrichment
+Will occur in the translate function, after reception, translation, and enrichment but right
+before dispatch.Verify that the final dataset meets configured receiver expectations prior to dispatch.
+
+When a message fails validation, validation errors should show up in:
+- The submission history API
+- The Action Log in the database in such a way that the engagement team can easily query for it
+  - Submission validation failures should show up in Submission History UI
+    - Sender is responsible for issue triage escalating to ReportStream as necessary
+  - Send failures should show up in last-mile failures
+    - ReportStream is responsible for issue triage
+- Notify the sender immediately in API response if parse validation fails
+
+Tickets needed for both:
+- Create the validation profile setting: https://github.com/CDCgov/prime-reportstream/issues/9161
+- Add validation errors to the submission history API: https://github.com/CDCgov/prime-reportstream/issues/9036
+- Add validation errors to the action log: https://github.com/CDCgov/prime-reportstream/issues/9221
+
+### FHIR Validation
+
+#### Validation Tool
+There is an existing and well-maintained project that meets the requirements for a validation tool. We intend to deploy
+that tool as an azure function for an HL7 validation web app.
+
+See the `FHIR Validator Wrapper` section under the `Background Information` heading below for more information.
+
+
+#### Background Information
+The HAPI FHIR library represents an implementation guide using:
+- [Structure Definitions](http://hl7.org/fhir/structuredefinition.html) - Valid fields, data types, cardinalities, valid values, and further rules.
+- [Code Systems](http://hl7.org/fhir/codesystem.html) - Code and vocabulary definitions
+- [Value Sets](http://hl7.org/fhir/valueset.html) - Lists of codes (from above) for use in field values
+
+The HAPI FHIR library supports three types of validation:
+- [Parser](https://hapifhir.io/hapi-fhir/docs/validation/parser_error_handler.html)
+  - Lightweight but not comprehensive
+  - An IParser instance can be configured with StrictErrorHandler or LenientErrorHandler resulting in exceptions or log messages upon parsing an invalid FHIR bundle.
+- [Instance](https://hapifhir.io/hapi-fhir/docs/validation/instance_validator.html)
+  - Heavy and comprehensive
+  - Great user-facing error messages
+  - Flexible and easy to write/configure
+  - Built-in support for core FHIR spec and many derivatives
+- [<s>Schema / Schematron</s>](https://hapifhir.io/hapi-fhir/docs/validation/schema_validator.html)
```

Comment:
Why is this crossed out? Perhaps just delete it and update the number above to say 2?

Reply:
@thetaurean can you please address this one?